stringfuzz/transformers/reverse.py

```diff
@@ -0,0 +1,32 @@
+'''
+Reversing every string literal
+'''
+
+import random
+import string
+
+from stringfuzz.ast import StringLitNode
+from stringfuzz.ast_walker import ASTWalker
+from stringfuzz.parser import parse
+
+__all__ = [
+    'reverse',
+]
+
+class ReverseTransformer(ASTWalker):
+    def __init__(self, ast):
+        super(ReverseTransformer, self).__init__(ast)
+
+    @property
+    def ast(self):
+        return self._ASTWalker__ast
+
+    def exit_literal(self, literal):
+        if isinstance(literal, StringLitNode):
+            literal.value = literal.value[::-1]
+
+# public API
+def reverse(s, language):
+    expressions = parse(s, language)
+    transformer = ReverseTransformer(expressions).walk()
+    return transformer.ast
```

Comment:
Good point about concats! I'll fix that.

As for the requirement on operators. I see this as a requirement for sat preservation, not a requirement on the transformation itself. That is, if we happen to give the transformer a problem without these operators then we will get a sat preserving transformation, if we don't then we won't and that's OK because we'll still be fuzzing the sovler.

Reply:
Yep, that's correct. By mentioning IndexOf, I meant to say that in order for the transformation to preserve satisfiability, that operator had to be absent, but also concat order had to be reversed. Thanks for the fix!